File: CSharp/DIP/Before.cs
// Low-level module
public class Database
{
    public void Save(string userData)
    {
        Console.WriteLine($"Saved to database: {userData}");
    }
}

// High-level module
public class UsersService
{
    private Database _database;

    public UsersService()
    {
        _database = new Database(); // Direct dependency on a concrete class
    }

    public void RegisterUser(string userData)
    {
        _database.Save(userData);
    }
}

//
//
// var userService = new UserService();
// userService.RegisterUser("John Doe");

File: CPlusPlus/DIP/Before.cpp
#include <iostream>
#include <string>

// Low-level module
class Database {
public:
    void Save(const std::string& userData) {
        std::cout << "Saved to database: " << userData << std::endl;
    }
};

// High-level module tightly coupled to Database
class UsersService {
private:
    Database database;

public:
    void RegisterUser(const std::string& userData) {
        database.Save(userData);
    }
};
